src/Symfony/Component/Messenger/Middleware/HandleMessageMiddleware.php

```
@@ -64,6 +67,10 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
 
                 /** @var AckStamp $ackStamp */
                 if ($batchHandler && $ackStamp = $envelope->last(AckStamp::class)) {
+                    if ($envelope->last(BusNameStamp::class) && self::PARALLEL_BUS === $envelope->last(BusNameStamp::class)->getBusName()) {
+                        throw new HandlerFailedException($envelope, [new LogicException("Parallel bus can't be used for batch messages")]);
```

I don't think we necessarily need to handle in this PR, but do you think it will be possible to support batch messages in parallel in the future? My project doesn't use batch messages today but may in the future.

---

If we need to support batch processing, it seems we'll need to make additional modifications. There are several considerations to bear in mind, particularly regarding potential issues with serialization. Additionally, we need to consider that it is launched concurrently and properly identify the groups and limits involved. But I'd be willing to give it a try once this approach is accepted.

---

Sorry. I dont really understand. 

Why do you need to throw an exception here?

---

Basically parallel message handling does not yet support batch message handling, so if you try to use the parallel bus and handle in batches, you get this exception.